example_progs.cpp

```diff
@@ -8008,6 +8008,96 @@ prog three_level_memory_two() {
   return prg;
 }
 
+prog three_level_pond() {
+  prog prg;
+  prg.compute_unit_file = "three_level_pond_compute.h";
+  prg.name = "three_level_pond";
+
+// Stencil<uint16_t, 34, 34> &input_host_stencil = arg_0;
+  prg.add_input("input_host_stencil");
+  prg.buffer_port_widths["input_host_stencil"] = 16;
+// Stencil<uint16_t, 32, 32> &hw_output_stencil = arg_1;
+  prg.add_output("hw_output_stencil");
+  prg.buffer_port_widths["hw_output_stencil"] = 16;
+
+////producing input_cgra.stencil
+  auto input_cgra_s0_y = prg.add_loop("input_cgra_s0_y", 0, 34);
+  auto input_cgra_s0_x = input_cgra_s0_y->add_loop("input_cgra_s0_x", 0, 34);
+
+//store is: input_cgra.stencil(input_cgra_s0_x, input_cgra_s0_y) = input_host.stencil(input_cgra_s0_x, input_cgra_s0_y)
+  auto hcompute_input_cgra_stencil = input_cgra_s0_x->add_op("op_hcompute_input_cgra_stencil");
+  hcompute_input_cgra_stencil->add_function("hcompute_input_cgra_stencil");
+  hcompute_input_cgra_stencil->add_load("input_host_stencil", "input_cgra_s0_y", "input_cgra_s0_x");
+  prg.buffer_port_widths["input_cgra_stencil"] = 16;
+  hcompute_input_cgra_stencil->add_store("input_cgra_stencil", "input_cgra_s0_y", "input_cgra_s0_x");
+
+//consuming input_cgra.stencil
+////producing output_cgra.stencil
+  auto output_cgra_s0_y_y_cgra = prg.add_loop("output_cgra_s0_y_y_cgra", 0, 4);
+  auto output_cgra_s0_x_x_cgra = output_cgra_s0_y_y_cgra->add_loop("output_cgra_s0_x_x_cgra", 0, 4);
+////producing hw_input.stencil
+  auto hw_input_s0_y = output_cgra_s0_x_x_cgra->add_loop("hw_input_s0_y", 0, 10);
+  auto hw_input_s0_x = hw_input_s0_y->add_loop("hw_input_s0_x", 0, 10);
+
+//store is: hw_input.stencil(hw_input_s0_x, hw_input_s0_y) = input_cgra.stencil(hw_input_s0_x, hw_input_s0_y)
+  auto hcompute_hw_input_stencil = hw_input_s0_x->add_op("op_hcompute_hw_input_stencil");
+  hcompute_hw_input_stencil->add_function("hcompute_hw_input_stencil");
+  hcompute_hw_input_stencil->add_load("input_cgra_stencil", "hw_input_s0_y", "hw_input_s0_x");
```

Comment:
I think this load operation from memory is not correct.  We still need the outer loop with stride.